VirtoCommerce.CatalogModule.Data/Search/Indexing/CategoryDocumentBuilder.cs

```diff
@@ -102,6 +103,16 @@ protected virtual IndexDocument CreateDocument(Category category)
                 document.AddObjectFieldValue(itemDto);
             }
 
+            document.Add(new IndexDocumentField("parent", category.ParentId ?? category.CatalogId) { IsRetrievable = true, IsSearchable = true, IsFilterable = true });
+
+            if (!category.Parents.IsNullOrEmpty())
```

Comment:
How the query  "return me all direct children of the given category" will look like? If you will store in an index array of keys for all parent categories?

Reply:
for example: 
![image](https://user-images.githubusercontent.com/8361000/73478588-06226580-439f-11ea-8617-e087fd3c5061.png)
make a request: give all direct children of 'Camcorders' category with Id: '45d3fc9a913d4610a5c7d0470558c4b2'
request body:
```
{
	"responseGroup": "withCategories, withProducts",
	"sort": "",
	"skip": 0,
	"take": 50,
	"terms": [ "parent:45d3fc9a913d4610a5c7d0470558c4b2"]
}
```
then in the result
```
{
  "totalCount": 2,
  "items": [
    {
      "parentId": "45d3fc9a913d4610a5c7d0470558c4b2",
      "isVirtual": false,
      "code": "33228",
      "catalogId": "4974648a41df4e6ea67ef2ad76d7bbd4",
      "name": "Aerial Imaging & Drones",
      "outline": "45d3fc9a913d4610a5c7d0470558c4b2",
      "path": "Camcorders",
      "isActive": true,
      "priority": 0,
      "links": [
        {
          "priority": 0,
          "catalogId": "1662f5e03af04275aa5d2653357d20e4",
          "categoryId": "64b5ba1277ab4649a6caf20f4e6f17c1"
        }
      ],
      "imgSrc": "http://localhost/admin/assets/catalog/33228/1419605414000_1106906.jpg",
      "images": [
        {
          "sortOrder": 0,
          "relativeUrl": "catalog/33228/1419605414000_1106906.jpg",
          "url": "http://localhost/admin/assets/catalog/33228/1419605414000_1106906.jpg",
          "typeId": "image",
          "group": "images",
          "name": "1419605414000_1106906.jpg",
          "isInherited": false,
          "id": "7406407cf0504464b53a9adc0e9bd297"
        }
      ],
      "seoObjectType": "Category",
      "seoInfos": [
        {
          "semanticUrl": "aerial-imaging-drones",
          "objectId": "e51b5f9eea094a44939c11d4d4fa3bb1",
          "objectType": "Category",
          "isActive": true,
          "languageCode": "en-US",
          "createdDate": "2015-10-06T22:39:19.047Z",
          "modifiedDate": "2015-10-06T22:39:19.047Z",
          "createdBy": "unknown",
          "modifiedBy": "unknown",
          "id": "42673c31224e46c4b51b4c026401e8f3"
        }
      ],
      "outlines": [
        {
          "items": [
            {
              "id": "4974648a41df4e6ea67ef2ad76d7bbd4",
              "seoObjectType": "Catalog",
              "seoInfos": [],
              "hasVirtualParent": false
            },
            {
              "id": "45d3fc9a913d4610a5c7d0470558c4b2",
              "seoObjectType": "Category",
              "seoInfos": [
                {
                  "semanticUrl": "camcorders",
                  "isActive": true,
                  "languageCode": "en-US",
                  "createdDate": "0001-01-01T00:00:00Z"
                }
              ],
              "hasVirtualParent": false
            },
            {
              "id": "e51b5f9eea094a44939c11d4d4fa3bb1",
              "seoObjectType": "Category",
              "seoInfos": [
                {
                  "semanticUrl": "aerial-imaging-drones",
                  "isActive": true,
                  "languageCode": "en-US",
                  "createdDate": "0001-01-01T00:00:00Z"
                }
              ],
              "hasVirtualParent": false
            }
          ]
        },
        {
          "items": [
            {
              "id": "1662f5e03af04275aa5d2653357d20e4",
              "seoObjectType": "Catalog",
              "seoInfos": [],
              "hasVirtualParent": false
            },
            {
              "id": "64b5ba1277ab4649a6caf20f4e6f17c1",
              "seoObjectType": "Category",
              "seoInfos": [
                {
                  "semanticUrl": "drones",
                  "isActive": true,
                  "languageCode": "en-US",
                  "createdDate": "0001-01-01T00:00:00Z"
                }
              ],
              "hasVirtualParent": false
            },
            {
              "id": "e51b5f9eea094a44939c11d4d4fa3bb1",
              "seoObjectType": "Category",
              "seoInfos": [
                {
                  "semanticUrl": "aerial-imaging-drones",
                  "isActive": true,
                  "languageCode": "en-US",
                  "createdDate": "0001-01-01T00:00:00Z"
                }
              ],
              "hasVirtualParent": true
            }
          ]
        }
      ],
      "createdDate": "2015-10-06T22:39:19.043Z",
      "modifiedDate": "2018-06-12T06:16:33.273Z",
      "createdBy": "unknown",
      "modifiedBy": "unknown",
      "id": "e51b5f9eea094a44939c11d4d4fa3bb1"
    },
    {
      "parentId": "45d3fc9a913d4610a5c7d0470558c4b2",
      "isVirtual": false,
      "code": "f7a8c",
      "catalogId": "4974648a41df4e6ea67ef2ad76d7bbd4",
      "name": "Consumer camcorders",
      "outline": "45d3fc9a913d4610a5c7d0470558c4b2",
      "path": "Camcorders",
      "isActive": true,
      "priority": 0,
      "links": [],
      "imgSrc": "http://localhost/admin/assets/catalog/f7a8c/1023206.jpg",
      "images": [
        {
          "sortOrder": 0,
          "relativeUrl": "catalog/f7a8c/1023206.jpg",
          "url": "http://localhost/admin/assets/catalog/f7a8c/1023206.jpg",
          "typeId": "image",
          "group": "images",
          "name": "1023206.jpg",
          "isInherited": false,
          "id": "dadd60f8d2b3498383f72b37814aef12"
        }
      ],
      "seoObjectType": "Category",
      "seoInfos": [
        {
          "semanticUrl": "consumer-camcorders",
          "objectId": "53e239451c844442a3b2fe9aa82d95c8",
          "objectType": "Category",
          "isActive": true,
          "languageCode": "en-US",
          "createdDate": "2015-10-06T22:39:19.1Z",
          "modifiedDate": "2015-10-06T22:39:19.1Z",
          "createdBy": "unknown",
          "modifiedBy": "unknown",
          "id": "de04cdb70d0145ea8b6c68e22e054cf2"
        }
      ],
      "outlines": [
        {
          "items": [
            {
              "id": "4974648a41df4e6ea67ef2ad76d7bbd4",
              "seoObjectType": "Catalog",
              "seoInfos": [],
              "hasVirtualParent": false
            },
            {
              "id": "45d3fc9a913d4610a5c7d0470558c4b2",
              "seoObjectType": "Category",
              "seoInfos": [
                {
                  "semanticUrl": "camcorders",
                  "isActive": true,
                  "languageCode": "en-US",
                  "createdDate": "0001-01-01T00:00:00Z"
                }
              ],
              "hasVirtualParent": false
            },
            {
              "id": "53e239451c844442a3b2fe9aa82d95c8",
              "seoObjectType": "Category",
              "seoInfos": [
                {
                  "semanticUrl": "consumer-camcorders",
                  "isActive": true,
                  "languageCode": "en-US",
                  "createdDate": "0001-01-01T00:00:00Z"
                }
              ],
              "hasVirtualParent": false
            }
          ]
        }
      ],
      "createdDate": "2015-10-06T22:39:19.097Z",
      "modifiedDate": "2018-06-12T06:16:33.273Z",
      "createdBy": "unknown",
      "modifiedBy": "unknown",
      "id": "53e239451c844442a3b2fe9aa82d95c8"
    }
  ]
}
```

Reply:
@tatarincev that's ok?